include/irr/asset/IDescriptorSet.h

```diff
@@ -39,19 +39,36 @@ class IDescriptorSet
     struct SDescriptorInfo
     {
         core::smart_refctd_ptr<IDescriptor> desc;
-        union {
```

Comment:
why no union?

Reply:
because it doesnt call destructors of its members (deleted destructor). User has to provide user-defined destructor deciding which destructors are called, but it's impossible to tell since `SDescriptorInfo` doesnt know descriptor type

Reply:
Ekhm, there's a `desc` just on the line above, could add a `getType()` function to it.

The you could switch/if in the destructor to decide which union to kill.

Would save us from a lot of `dynamic_cast`ing as well.


### Sidenote on performance

To get a performance increase, it will make sense to re-use the memory in `SWriteDescriptorSet::info` (so Vulkan backend doesn't need to allocate arrays of `VkDescriptorImageInfo`, `VkDescriptorBufferInfo` or `VkBufferView`)

Hence `IDescriptorSet::updateDescriptorSet` should **consume** the entire `_descWrites` parameter and destroy the `SWriteDescriptorSet` objects inside, so `SWriteDescriptorSet::info` should come back from the function as just garbage memory ready to be deallocated.

Reply:
>Ekhm, there's a desc just on the line above, could add a getType() function to it.

we cant because one irrbaw class doesnt map to one particular desc type. But what we can do is a function telling "descriptor category" - i.e. what union member to use.

Reply:
>Sidenote on performance

i dont get it at all

Reply:
> i dont get it at all 

Nvm leave it till later.

Reply:
> can do is a function telling "descriptor category" - i.e. what union member to use.

Let's do that.